script.js

```diff
@@ -1,4 +1,324 @@
+// more comfortable features included:
+// Score, Leaderboard, Lowest Combined Mode, Auto-Generate Mode, Variable Number of Dice
+// Score and Leaderboard are generated for every game, BUT Leaderboard is not sorted
+// v-Dice only works with auto, score calculation is broken if not flagged
+// all modified modes are activated with modifier flags
+// multiple flags can be used at a time (see below)
+// v-Players framework exists, but is not implemented
+// knockout is completely not implemented
+
+// variables for MVP- 2 player manual
+var activePlayer = 1;
+var currentGameProgressMode = "initializing";
+var currentPlayerArray = [0, 0];
+var currentScore = 0;
+var scoreArray = [[], []];
+// ^this is meant to be a nested array - index 0 corresponds to player 1's scores, etc.
+// initialize later in main
+var progressNumber = 0; // incremented through player turns until a round is complete
+const validCommands = ["start", "Roll", "dice 1", "dice 2", "next"];
+// ^ modes must be applied with "--" modifiers, e.g. --lowestCombined
+// a full command might look like start --auto
+
+// variables for MC - modifiers
+var scoringModeIsHighest = true;
+var diceChoiceSelectionSkip = false;
+var numberOfPlayers = 2; // default
+var numberOfDice = 2; // default
+const modifierMessages = [
+  "You have applied the 'lowestCombined' modifier. For this game, the player with the smaller number wins each round.<br>",
+  "You have applied the 'auto' modifier. The best combination for your chosen scoring rules will always be applied.<br>",
+  "You have applied the 'vDice' modifier. Note that you MUST use this modifier with the 'auto' modifier.<br>",
+  "You have entered an invalid modifier. Only --lowestCombined, --auto, and --vDiceN are accepted. Please try again.<br>",
+];
+
+var validateInput = function (input) {
+  var parsedInput = "";
+  if (input.includes("dice") || input.includes(" ") == false) {
+    parsedInput = input;
+  } else if (input.includes(" ") == true) {
+    parsedInput = input.substr(0, input.indexOf(" "));
+    // check only substring before the first space
+  }
+  // console.log(`parsedInput: ${parsedInput}`);
+  if (validCommands.includes(parsedInput)) {
+    return true;
+  }
+  // console.log(input);
+  // console.log("Invalid Input.");
+  return false;
+};
+
+var createArrayFromString = function (str) {
+  var arr = str.split(" ");
+  var removedElement = arr.shift(); // remove the first entry, which should ALWAYS be "start"
+  console.log(`removed ${removedElement}`);
+  return arr;
+};
+
+var parseVariableDiceModifier = function (vdicemdf) {
+  // vdicemdf should take the form "--vdiceX", where X is an integer
+  // the "--vdice" substring occupies the 0th position to the 6th position
+  var customNumberOfDice = vdicemdf.substring(7); // remove "--vdice" and read remaining numbers
+  customNumberOfDice = Number(customNumberOfDice);
+  numberOfDice = customNumberOfDice;
```

Comment:
Is it possible to have just done numberOfDice = Number(customNumberOfDice) ?

Reply:
yes ;-;